### ISSUE TITLE
FileChooser accept for file upload not working when the request is intercepted to upload certificate details

### ISSUE THREAD

**Issue by arvindmr** [puppeteer/puppeteer#5179]
### Steps to reproduce
Requirement:
With puppeteer a file upload of type (CSV) needs to be uploaded to a site by crawling the site.
The site under test is a HTTPS enabled site. Issue described below.

API used is fileChooser.accept(). This works when the request interception for SSL certificate is fed manually, while on the other hand, the fileChooser gets stuck indefinitely when the requestInterception send the certificate information as part of request(options). Note that, once the fileChooser.accept() is invoked, the site makes an Asynchronous request to its server which expects the SSL certificate information.

Note that i have tried all the other suggestions of ignoreHTTPSErrors to true, ignore-certificate-errors flag etc, but nothing is much helpful.

**Tell us about your environment:**

* Puppeteer version:
2.0.0
* Platform / OS version:
Ubuntu 18.04
* URLs (if applicable):
* Node.js version:
v8.16.0

**What steps will reproduce the problem?**
Puppeteer config
```
 puppeteer: {
        launch_options: {
        slowMo: 20,
        args: [
            '--disable-setuid-sandbox',
            '--no-sandbox'
        ],
        executablePath: 'node_modules/puppeteer/.local-chromium/linux-706915/chrome-linux/chrome',
        headless: false,
        }
    }
```
_Please include code that reproduces the issue._

Following code works if the request is intercepted manually for allowing the Certificates to be chosen. i.e when the browser automatically pops up the certificate dialog.
```
const uploadCSV = (page, credentials) => {
    return new Promise(async (resolve, reject) => {
        await page.setRequestInterception(true);
        await page.on('request', interceptedRequest => {
            interceptedRequest.continue();
        });
        await page.waitForSelector('input[name="Upload File"]');
        const [fileChooser] = await Promise.all([
            page.waitForFileChooser({ timeout: 60000 }),
            page.click('input[name="Upload File"]'),
        ]);

        try {
            await fileChooser.accept(['./content.csv']);
        } catch (e) {
            resolve({
                isError: true,
                message: e.message
            });
        }
        page.waitFor(30000).then(() => {
            resolve();
        }, () => {
            reject();
        });
    });
}
```
**Actual Issue:** The following code gets stuck indefinitely, when the fileChooser accept is invoked. In-spite of intercepting the requests and sending the certificates.
```
uploadCSV = (page, config) => {
return new Promise(async (resolve,reject) => {
await page.setRequestInterception(true);

        page.on('request', interceptedRequest => {
            var options = {
                uri: interceptedRequest.url(),
                method: interceptedRequest.method(),
                headers: interceptedRequest.headers(),
                body: interceptedRequest.postData(),
                pfx: fs.readFileSync(config.certificate_path),
                passphrase: config.certificate_passphrase,
            };
            request(options, function (err, resp, body) {
                if (err) {
                    return interceptedRequest.abort('connection refused');
                }
                interceptedRequest.respond({
                    status: resp.statusCode,
                    contentType: resp.headers['content-type'],
                    headers: resp.headers,
                    body: body
                });
            });
         });

await page.waitForSelector('input[name="Upload File"]');
        const [fileChooser] = await Promise.all([
            page.waitForFileChooser({timeout:60000}),
            page.click('input[name="Upload File"]'),
        ]);

 try {
        await fileChooser.accept(['./content.csv']);     
        } catch (e) {
                resolve({
                    isError: true,
                    message: e.message
                });
            }
       page.waitFor(30000).then(() => {
            resolve();
        }, () => {
            reject();
        });
}
```
**What is the expected result?**
The file chooser.accept should return successfully after the server returned the Asynchronous response for the upload file. As outlined above, the steps when the interceptRequest.continue is given, the cert information flows properly to the server and the file upload is successful. 

**What happens instead?**
Either the request interception is stuck or the fileChooser is not able to send the Asynchrounous Response with the right set of intercepted SSL requests (when the intercepted request is filled with the certificate information as part of the request options) that the file upload gets stuck indefinitely.

Any help or work around or any options to try and overcome this one the interception of requests is much appreciated.

Thanks
Arvind

**Comment by kblok** [comment]
Could you improve the code formatting? It's hard to follow in this way.

**Comment by arvindmr** [comment]
@kblok - Yeah taken care. Now its lot better.
